dbuf-lsp/src/core/navigator/indentify.rs

```diff
@@ -0,0 +1,216 @@
+//! Indetifies symbol at location and returns it type.
+//!
+
+use tower_lsp::lsp_types::Position;
+
+use crate::core::ast_access::{
+    ElaboratedAst, ElaboratedHelper, LocStringHelper, LocationHelpers, Str,
+};
+
+use crate::core::ast_visitor::scope_visitor::ScopeVisitor;
+use crate::core::ast_visitor::VisitResult::*;
+use crate::core::ast_visitor::*;
+
+use super::Navigator;
+use super::Symbol;
+
+struct GetImpl<'a> {
+    elaborated: &'a ElaboratedAst,
+    target: Position,
+    scope: ScopeVisitor<'a>,
+    result: Symbol,
+}
+
+pub fn get_symbol_impl(navigator: &Navigator, pos: Position) -> Symbol {
+    let mut implementation = GetImpl {
+        elaborated: navigator.elaborated,
+        target: pos,
+        scope: ScopeVisitor::new(navigator.elaborated),
+        result: Symbol::None,
```

Comment:
You can add the type parameter to the `Visitor` trait to allow for it to return a result during the visit.